tests/unit/test_charm.py

```diff
@@ -78,26 +78,39 @@ def test_on_config_changes_aborts_if_host_is_container(harness, mocker):
 
 
 @pytest.mark.parametrize(
-    "initiator_content, expected_initiator_name",
+    "initiator_content, expected_initiator_name, initiatorname_file",
```

Comment:
What does initiator_name_file mean? It mixed the value "True" and "None" together.

Reply:
It's a missing part in unit test.


There is a logic that `_get_initiator_name_from_file` function can take `None` as `initiatorname_file` input value. So Ture means using the mock value for it and `None` means pass `None` to the function.

Reply:
Thanks for adding an extra test case. But I think we could use some nicer variable names and clean the logic up a bit.

```python
@pytest.mark.parametrize(
    "initiator_content, expected_initiator_name, iscsi_config_file_provided",
    [
        (None, None, True),  # file doesn't exist
        ("", None, True),  # file exists but no content
        (INITIATOR_CONTENT_TEMPLATE.format("InitiatorName=test-iqn"), "test-iqn", True),
        (INITIATOR_CONTENT_TEMPLATE.format("InitiatorName=test-iqn"), "test-iqn", False),
    ...
)
    ...
    harness.charm.ISCSI_CONF_PATH = tmp_path
    if iscsi_config_file_provided:
        initiatorname_file = tmp_path / "initiatorname.iscsi"
        if initiator_content is not None:
            initiatorname_file.write_text(initiator_content)
    else:
        harness.charm.ISCSI_INITIATOR_NAME.write_text(initiator_content)
    initiator_name = harness.charm._get_initiator_name_from_file(initiatorname_file)
    assert initiator_name == expected_initiator_name
```

* `initiatorname_file` --> `iscsi_config_file_provided` to more easily understand that this denotes whether the arg to provide the initiatorname.iscsi config file was provided or not.
* Multiple if clauses was simplified to one if/else clause.

Reply:
Updated.